code/modules/cm_aliens/XenoStructures.dm

```diff
@@ -375,6 +375,11 @@
 		if (L.hivenumber == hivenumber)
 			L.scuttle(src)
 		return
+	if(isfacehugger(user))
+		var/mob/living/carbon/xenomorph/facehugger/F = user
+		if (F.hivenumber == hivenumber)
+			F.scuttle(src)
+		return
 	if(iscarbon(user))
 		var/mob/living/carbon/C = user
 		if (C.ally_of_hivenumber(hivenumber))
```

Comment:
I'd suggest changing this proc in general so that rather than typechecking larva or facehuggers, it does a mob_size check. 
Either way, if you don't want to do that, you need to change `islarva(user)` to `islarva(user) || isfacehugger(user)` and change `var/mob/living/carbon/xenomorph/larva/L = user` to ``var/mob/living/carbon/xenomorph/crawler` to account for it no longer being a larva exclusively.
No need to write the same code out twice.

Reply:
Thoughts on doing it like this:
```
mob/living/carbon/xenomorph/proc/scuttle
      return
```

```
  if(isxeno(user))
     var/mob/living/carbon/xenomorph/X = user
     if (user.mob_size == MOB_SIZE_SMALL && X.hivenumber == hivenumber)
           X.scuttle(src)
    return
```

There's no problem with checking mob_size but the typecasting is what I'm worried about. My first iteration was exactly just adding "|| isfacehugger(user)"

But then it typecasts it to a larva. Seems to work similarly since their scuttle code is the exact same so we could always do it like that. An issue would only occur if huggers and larva scuttle code ever differed.

Reply:
I refactored it. Removed the duplicate, placed it on the parent level and handled it where it only affects xenos with the proper mob size. All xenos can scuttle but it does nothing other than return false for larger xenos. The boolean value is to ensure if they could properly scuttle